### DISCUSSION TITLE
Can't open serial port /dev/zwave

### DISCUSSION THREAD

**Discussion by sstainba** [zwave-js/zwave-js-ui#2002] [Q&A]
I am attempting to set up zwavejs2mqtt on TrueNAS Scale.  I have a Zooz zwave stick and in the container settings I have it mapped (by-id) to /dev/zwave.  The container is running as the "apps" user (no sudo privs) but the container is privileged.  However, I keep getting an error in the web ui that says the driver failed to open the serial port (permission denied).  Any ideas what I can check/change to get through this?

**Comment by robertsLando** [comment]
Does the user that is running the container has the permissions to access the serial device? If you could share your docker compose file it could be easier to debug this. 
cc @kpine @blhoward2

**Comment by blhoward2** [comment]
This is almost certainly a permissions issue. Either your docker instance can't access the by-id device file, or the process can't access the /dev/zwave one inside the container.

Make sure your docker user is in the dialout group on the host. I agree that seeing your docker compose would help.

**Comment by kpine** [comment]
I don't know how containers work on TrueNAS Scale, so another possible problem could be the controller soft-reset. You can try disabling that in the Z-Wave settings. We need more information to provide any help.

https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/app_crash
https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=driver-logs

**Comment by yakclock** [comment]
I'm having a similar problem. I'm running docker as rootless with the user 'ubuntu' which is in the dialout group.

'Silly' Level logs are below. I'm running from a Makefile like this:

`
docker run -d --name zwavejs2mqtt -p 8091:8091 -p 3000:3000 --device /dev/serial/by-id/usb-0658_0200-if00 -v /srv/containers/sandpit/zwavejs2mqtt/store:/usr/src/app/store zwavejs/zwavejs2mqtt:latest
`



`
2021-12-05T10:22:05.471Z DRIVER   version 8.8.3
2021-12-05T10:22:05.471Z DRIVER
2021-12-05T10:22:05.472Z DRIVER   starting driver...
2021-12-05T10:22:05.481Z DRIVER   opening serial port /dev/serial/by-id/usb-0658_0200-if00
2021-12-05T10:22:14.522Z DRIVER   Failed to open the serial port: Error: Permission denied, cannot open /dev/ser
                                  ial/by-id/usb-0658_0200-if00
2021-12-05T10:22:14.523Z DRIVER   destroying driver instance...
2021-12-05T10:22:14.527Z DRIVER   driver instance destroyed
`

Could this be because its rootless docker ? I know that there are limitations when running rootless but I cannot find anything that specifically mentions serial devices.

**Reply by robertsLando** [reply]
It could be for sure

**Comment by jrgleason** [comment]
I am running mine native on Raspberry Pi OS with `npm run start` and I am getting the same error. I tried with `sudo` and I also tried to add the pi user to dialout but it already was. Any ideas would be great. I also tried using `su` to root and running from that.

**Reply by kpine** [reply]
Why are you using npm?

https://zwave-js.github.io/zwave-js-ui/#/getting-started/other-methods?id=nodejs-or-pkg-version

**Reply by jrgleason** [reply]
> Why are you using npm?
> 
> https://zwave-js.github.io/zwave-js-ui/#/getting-started/other-methods?id=nodejs-or-pkg-version

I want to and actually the problem ended up being me running under root and it didn't have dialout (both sudo and su root) so I think I finally have it working but now the zwave server is booting up and I can see valid message (temp etc) but the UI never starts.

**Reply by jrgleason** [reply]
> > Why are you using npm?
> > https://zwave-js.github.io/zwave-js-ui/#/getting-started/other-methods?id=nodejs-or-pkg-version
> 
> I want to and actually the problem ended up being me running under root and it didn't have dialout (both sudo and su root) so I think I finally have it working but now the zwave server is booting up and I can see valid message (temp etc) but the UI never starts.

NM fixed that too
